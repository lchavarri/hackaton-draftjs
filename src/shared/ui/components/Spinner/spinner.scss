@import '../../assets/variables';

$spinner-md: 1.5rem;
$spinner-sm: 0.857142rem;

.spinner {
  /* Spinner size and color */
  width: $spinner-md;
  height: $spinner-md;
  border-color: #444;

  /* Additional spinner styles */
  animation: spinner 1s linear infinite;
  border-right-color: transparent;
  border-style: solid;
  border-width: 2px;
  border-radius: 50%;
  box-sizing: border-box;
  display: inline-block;
  vertical-align: middle;
  margin: 10px;

  &.primary {
    border-color: $hover;
    border-right-color: transparent;
  }

  &.sm {
    width: $spinner-sm;
    height: $spinner-sm;
  }

  &.fit {
    width: 100%;
    height: 100%;
    margin: 0;
  }
}

/* Animation styles */
@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
