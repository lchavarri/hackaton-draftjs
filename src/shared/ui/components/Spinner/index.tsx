import React from 'react';
import './spinner.scss';

export default function Spinner(props: any) {
  const { primary, size } = props;
  const variantClass = primary ? 'primary' : '';
  const sizeClass = size ? size : 'md';

  return <span className={`spinner ${variantClass} ${sizeClass}`} />;
}
