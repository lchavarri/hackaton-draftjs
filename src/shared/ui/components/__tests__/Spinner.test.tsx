import React from 'react';
import { shallow, mount, ReactWrapper } from 'enzyme';
import Spinner from '../Spinner';

describe('Spinner component', () => {
  let component: ReactWrapper;
  let props: any;

  const defaultProps = {};
  const getComponent = () => <Spinner {...props} />;
  const wrap = () => shallow(getComponent());
  const wrapDeep = () => mount(getComponent());

  beforeEach(() => {
    props = defaultProps;
    component = wrap();
  });

  it('should match component snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
