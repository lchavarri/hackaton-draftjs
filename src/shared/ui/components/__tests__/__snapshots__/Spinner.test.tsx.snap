// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should match component snapshot 1`] = `
<span
  className="spinner  md"
/>
`;
